File: players.py
import json
import constants as c

class players:
  def __init__(self):

    # Now lets open the season schedule for reference
    json_players = open(c.PLAYERS_PATH, "r")
    self.players = json.load(json_players)['Data']
    json_players.close()
    
  def getPlayerById(self, bowlerID):
    return next((item for item in self.players if item['BowlerID'] == bowlerID), None)
    
  def getPlayerByName(self, name):
    return next((item for item in self.players if item['BowlerName'] == name), None)
  
  def getPlayerNames(self):
    names = []
    
    for player in self.players:
      names.append(player['BowlerName'])
    return names
      
  def getGender(self, name):
    return self.getPlayerByName(name)['Gender']
    

File: www/js/players.js
class players {
    json_players = null;

    constructor () {
        return new Promise(resolve => {
            Promise.all([
                makeRequest("GET", PLAYERS_PATH).then(responseText => this.json_players = JSON.parse(responseText).Data),
                makeRequest("GET", FRIENDLY_NAMES_PATH).then(responseText => this.friendly_names = JSON.parse(responseText).Data)
            ]).then(() => {
                resolve(this);
            });
        });
    }

    getPlayerById(bowlerID) {
        return this.json_players.find(bowler => bowler.BowlerID == bowlerID);
    }

    getPlayerByName(name) {
        return this.json_players.find(bowler => bowler.BowlerName == name);
    }

    getPlayerNames() {
        return this.json_players.map(bowler => bowler.BowlerName);
    }
    
    getPlayerNamesByGender(gender) {
        return weeklyStandings.bowlerGames.players.getPlayerNames()
            .map(bowlerName => (weeklyStandings.bowlerGames.players.getGender(bowlerName) == gender) ? bowlerName : undefined)
            .filter(bowlerName => bowlerName != undefined);
    }

    getGender(name) {
        return this.getPlayerByName(name).Gender;
    }

    prettyName(bowlerName) {
        let name = this.friendly_names.find(bowler => bowler.BowlerName == bowlerName);

        if (undefined == name) {
            name = bowlerName;
        } else {
            name = name.FriendlyName;
        }

        return `${name.split(",")[1]} ${name.split(",")[0]}`.trim();
    }
}